packages/@react-aria/datepicker/src/useDatePickerGroup.ts

```diff
@@ -104,3 +129,19 @@ export function useDatePickerGroup(state: DatePickerState | DateRangePickerState
 
   return mergeProps(pressProps, {onKeyDown});
 }
+
+function orderSegments(editableSegments: NodeListOf<Element> | undefined) {
+  if (editableSegments) {
+    let segments = Array.from(editableSegments);
+    let segmentArr = segments.map(node => {
+      return {
+        element: node as FocusableElement,
+        rectX: node?.getBoundingClientRect().left
```

Comment:
if you run segments.filter(Boolean).map..., you may not need this `?`

might need to do it the line above so you can set the type correctly
```
let segments = Array.from(editableSegments).filter...
```